src/main/java/com/kodilla/ecommercee/service/GroupService.java

```diff
@@ -0,0 +1,36 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.Group;
+import com.kodilla.ecommercee.mapper.ProductMapper;
+import com.kodilla.ecommercee.repository.GroupRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class GroupService {
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Autowired
+    private ProductMapper productMapper;
+
+    public List<Group> getAllGroup() {
+        return groupRepository.findAll();
+    }
+
+    public Group saveGroup(final Group group) {
+        return groupRepository.save(group);
+    }
+
+    public void deleteGroup(final Long id) {
+        groupRepository.deleteById(id);
+    }
+
+    public Group getGroupById(final Long groupId) {
```

Comment:
Jak by Ci sie chcialo mozesz tutaj dodac GroupNotFoundException. Zobacz jak zrobilem to w klasie UserNotFoundException i wtedy robisz `.orElseThrow(GroupNotFoundException::new)`